### PR TITLE
HZC-2040: Add 'jsonSerializationExample' to the 5.0 client sample

### PR DESCRIPTION
**Pull request by sdreger**
The console output:

image

### CODE REVIEW

File: client.js [hunk 1]
```diff
@@ -83,6 +83,135 @@ async function sqlExample(hzClient) {
...
+
+async function populateCountriesMap(hzClient) {
+    // see: https://docs.hazelcast.com/hazelcast/5.0/data-structures/creating-a-map#writing-json-to-a-map
+    console.log("Populating 'countries' map with JSON values...");
+
+    const countries = await hzClient.getMap("country");
+    await countries.set("AU", {"isoCode":"AU", "country":"Australia"});
+    await countries.set("EN", {"isoCode":"EN", "country":"England"});
+    await countries.set("US", {"isoCode":"US", "country":"United States"});
+    await countries.set("CZ", {"isoCode":"US", "country":"Czech Republic"});
```
Comment (ivan-shevtsov) [inline client.js]: iso_code should be CZ

**Review by ivan-shevtsov — APPROVED** [review]

File: client.js [hunk 2]
```diff
@@ -83,6 +83,135 @@ async function sqlExample(hzClient) {
...
+    await selectCountriesAndCities(hzClient);
+
+    process.exit(0);
+}
+
+async function createMappingForCountries(hzClient) {
+    // see: https://docs.hazelcast.com/hazelcast/5.0/sql/mapping-to-maps#json-objects
+    console.log("Creating mapping for countries...");
+
+    const mappingQuery =
```
Comment (sergeyshakhov) [inline client.js]: Can we replace this with multiline string in order to improve readability?
```
    const mappingQuery = `
        CREATE OR REPLACE MAPPING country(
         __key VARCHAR,
         isoCode VARCHAR,
         country VARCHAR)
         TYPE IMap
         OPTIONS (
             'keyFormat' = 'varchar',
             'valueFormat' = 'json-flat'
         )
    `
```

Comment (sergeyshakhov) [reply client.js]: What js version are we aiming for?

Comment (sdreger) [reply client.js]: What I figured out from here: https://github.com/hazelcast/hazelcast-nodejs-client/blob/master/package.json
```
"engines": {
    "node": ">=10.4.0"
}
```

Comment (sergeyshakhov) [reply client.js]: But there is no NodeJS version mentioned in https://github.com/hazelcast/hazelcast-cloud-nodejs-sample-client/blob/master/package.json
Is it how it should be?

File: client.js [hunk 3]
```diff
@@ -83,6 +83,135 @@ async function sqlExample(hzClient) {
...
+    console.log("--------------------");
+}
+
+async function selectCitiesByCountry(hzClient, country) {
+    const query = "SELECT city, population FROM city where country=?";
+    console.log("Select city and population with sql = " + query);
+
+    const sqlResult = await hzClient.getSql().execute(query, [country]);
+    for await (const row of sqlResult) {
+        console.log("city = %s, population = %s", row.city, row.population);
```
Comment (sergeyshakhov) [inline client.js]: More readable alternative:
```
console.log(`city = ${row.city}, population = ${row.population}`);
```

File: client.js (hunk 2 above)
Comment (sergeyshakhov) [reply client.js]: In any case node 10.4 supports template literals:
https://node.green/#ES2015-syntax-template-literals
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Comment (sdreger) [reply client.js]: > 

Not sure it should be in our package.json. I can add it, but we need to know if it's needed here from some Node experts. @mdumandag WDYT?

Comment (sdreger) [reply client.js]: Fixed multiline strings.

File: client.js (hunk 1 above)
Comment (sdreger) [reply client.js]: Fixed, thanks.

File: client.js (hunk 3 above)
Comment (sdreger) [reply client.js]: Fixed as well. Thanks.

File: client.js (hunk 2 above)
Comment (mdumandag) [reply client.js]: Yes, we need at least Node 10 to run the client, so the applications that are using the client should be at least as new as this. So, I would also suggest adding that constraint here

Comment (sdreger) [reply client.js]: Thanks, added.

**Review by sergeyshakhov — APPROVED** [review]

**Pull request merged**